java/src/org/openqa/selenium/remote/AbstractDriverOptions.java

```diff
@@ -30,13 +35,47 @@
 import org.openqa.selenium.UnexpectedAlertBehaviour;
 import org.openqa.selenium.internal.Require;
 
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import java.util.TreeMap;
 import java.util.TreeSet;
 
 public abstract class AbstractDriverOptions<DO extends AbstractDriverOptions> extends MutableCapabilities {
+  public DO setBrowserVersion(String browserVersion) {
+    setCapability(
+      BROWSER_VERSION,
+      Require.nonNull("Browser version", browserVersion));
+    return (DO) this;
+  }
+
+  public DO setPlatformName(String platformName) {
+    setCapability(
+      PLATFORM_NAME,
+      Require.nonNull("Platform Name", platformName));
+    return (DO) this;
+  }
+
+  public DO setTimeouts(Map<String, Duration> timeouts) {
```

Comment:
Prefer a `setImplicitTimeout`, `setPageLoadTimeout` and `setScriptTimeout`, and then we don't need to do any validation, and it's really hard to get wrong.

Reply:
Much better implementation.